runtime/compiler/env/VMJ9.cpp

```diff
@@ -1294,6 +1294,69 @@ TR_J9VMBase::getObjectClassInfoFromObjectReferenceLocation(TR::Compilation *comp
    return ci;
    }
 
+TR::KnownObjectTable::ObjectInfo
+TR_J9VMBase::getObjClassInfoFromKnotIndex(TR::Compilation *comp, TR::KnownObjectTable::Index knotIndex)
+   {
+   TR::KnownObjectTable *knot = comp->getKnownObjectTable();
+   // Check to see whether the known object entry already holds the desired info.
+   // If so, return the cached version. Otherwise, fetch the info and cache it.
+   TR::KnownObjectTable::ObjectInfo existingObjInfo = knot->getObjectInfo(knotIndex);
+   TR::KnownObjectTable::ObjectInfo retrievedObjInfo;
+   TR::KnownObjectTable::ObjectInfo answerObjInfo;
+   TR_ASSERT_FATAL(existingObjInfo._jniReference, "The jniReference is missing for knowIndex %d", knotIndex);
+   bool infoIsCached = existingObjInfo._clazz != NULL;
+   if (comp->getOption(TR_DisableKnownObjectTableCaching) ||
+       !infoIsCached ||
```

Comment:
I think `comp->getOption(TR_DisableKnownObjectTableCaching)` should imply `!infoIsCached`, since the option will prevent the cache from being populated. It doesn't necessarily mean that a change is needed here, but we could potentially have a slight simplification, or alternately maybe an assertion